### DISCUSSION TITLE
Use `clustermq` with HPC that requires interactive session

### DISCUSSION THREAD

**Discussion by Aariq** [mschubert/clustermq#292] [Q&A]
My current HPC requires an interactive session to load R (`module load R` doesn't work on login node).  Is there a way to get `clustermq` working?  They provide a nice shortcut `interactive` that does:
```
salloc --job-name=interactive --mem-per-cpu=4GB --nodes=1  --ntasks=1 --time=01:00:00 --account=windfall --partition=windfall
```

Could I add `interactive` somewhere to either my ~/.ssh/config or in ssh_clustermq.tmpl, and then add an appropriately long timeout to get this working?  I haven't had success so far, and I just wanted to see if this was even a possibility before trying much more.

**Comment by mschubert** [comment]
I'm assuming you are referring to using `clustermq` via SSH because you mentioned the log/template files. You can do that in the *SSH.tmpl* file. The line that starts R:

https://github.com/mschubert/clustermq/blob/78125771cda27373bf19eb0f38ac5106524d4eae/inst/SSH.tmpl#L5

will perform these steps if you modify it to something like

```r
    "srun ..parameters.. module load R && R --no-save --no-restore -e \
```

(I *think* you need *srun* instead of *salloc* here)

Alternatively, if you have an interactive node that you can connect to without a job allocation and would like to use that instead, you can [specify this host in your *~/.ssh/config* via ProxyJump](https://www.redhat.com/sysadmin/ssh-proxy-bastion-proxyjump#:~:text=ProxyJump%20is%20the%20simplified%20way,the%20proxy%20or%20bastion%20hosts.).
